### PR TITLE
feat(ediscovery): support decryption of meeting activities

### PR DESCRIPTION
**Pull request by gannons**
Adding support for the decryption of meeting activities. Specifically 

Meeting Titles

    {
      "activityId": "65d2c570-fe42-11ea-a10e-81bb18766f0a",
      "timestamp": "2020-09-24T08:46:08.000Z",
      "verb": "add",
      "actorId": "926423f2-4b7f-4654-94e0-f05faa311be4",
      "actorDisplayName": "compliance@yopmail.com",
      "targetId": "894a4458af354ebfbf3fa1c025fb88b8_I_172954877251558750",
      "meeting": {
        "id": "894a4458af354ebfbf3fa1c025fb88b8_I_172954877251558750",
        "objectType": "meeting",
        "title": "XXXXXXXXX",
        "start": "2020-09-24T08:39:49.000Z",
        "end": "2020-09-24T08:46:08.000Z",
        "webLink": "https://publictest.dmz.webex.com/webappng/sites/publictest/meeting/postinfo/894a4458af354ebfbf3fa1c025fb88b8_I_172954877251558750"
      },
      "spaceName": "compliance test's Personal Room",
      "containerName": "compliance test's Personal Room"
    }

Recording Topics

    {
      "activityId": "c60367b0-fe42-11ea-a10e-81bb18766f0a",
      "timestamp": "2020-09-24T08:48:50.000Z",
      "verb": "add",
      "actorId": "926423f2-4b7f-4654-94e0-f05faa311be4",
      "actorDisplayName": "compliance@yopmail.com",
      "targetId": "894a4458af354ebfbf3fa1c025fb88b8_I_172954877251558750",
      "recording": {
        "id": "e0cc39b345a14ac9b9c3e9fcdfc4665f",
        "objectType": "recording",
        "downloadUrl": "https://integration.webexapis.com/v1/recordings/e0cc39b345a14ac9b9c3e9fcdfc4665f",
        "topic": "XXXXXXX",
        "durationSeconds": 21,
        "sizeBytes": 122506,
        "shareToMe": false
      },
      "spaceName": "compliance test's Personal Room",
      "containerName": "compliance test's Personal Room"
    }

### CODE REVIEW

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js [hunk 1]
```diff
@@ -248,6 +249,36 @@ class Transforms {
             }));
         }
 
+        // Decrypt meeting title if present
+        if (activity.meeting && activity.meeting.title) {
```
Comment (mwegman) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js:253]:
```suggestion
        if (activity.meeting?.title) {
```

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js [hunk 2]
```diff
@@ -248,6 +249,36 @@ class Transforms {
...
+            .catch((reason) => {
+              ctx.webex.logger.error(`Decrypt activity.meeting.title error for activity ${activity.activityId} in container ${activity.targetId}: ${reason}`);
+              activity.error = reason;
+
+              return Promise.resolve(object);
+            }));
+        }
+
+        // Decrypt meeting recording topic if present
+        if (activity.recording && activity.recording.topic) {
```
Comment (mwegman) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js:268]:
```suggestion
        if (activity.recording?.topic) {
```

**Review by mwegman — APPROVED** [review]

**Review by lalli-flores — APPROVED** [review]
It would be good to add unit tests for fail scenarios

File: packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js [hunk 3]
```diff
@@ -248,6 +249,36 @@ class Transforms {
             }));
         }
 
+        // Decrypt meeting title if present
+        if (activity?.meeting?.title) {
```
Comment (sregger) [inline packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js:253]: I've wanted something like Optional Chaining for a long time but could not find it. Thanks for the tip @mwegman.

File: packages/node_modules/@webex/internal-plugin-ediscovery/test/unit/spec/transforms.js [hunk 4]
```diff
@@ -244,6 +246,20 @@ describe('EDiscovery Transform Tests', () => {
...
 
+      it('Decrypt function throws an exception', () => {
+        ctx.webex.internal.encryption.decryptText = sinon.stub().returns(Promise.reject(new Error('@@@@@@')));
+
+        const result = Transforms.decryptReportContent(ctx, object, reportId)
+          .then(() => {
+            assert.fail('Decrypt did not fail as expected');
+          })
+          .catch(() => {
+            assert.isDefined(activity.error);
```
Comment (sregger) [inline packages/node_modules/@webex/internal-plugin-ediscovery/test/unit/spec/transforms.js:257]: Adding a check for an error scenario. Other scenarios such as `onBehalfOf` being undefined are already in place.

**Pull request merged**